src/live_restore/live_restore_server.c

```diff
@@ -155,10 +159,22 @@ __live_restore_worker_run(WT_SESSION_IMPL *session, WT_THREAD *ctx)
     WT_CURSOR *cursor;
     WT_SESSION *wt_session = (WT_SESSION *)session;
     ret = wt_session->open_cursor(wt_session, work_item->uri, NULL, NULL, &cursor);
+    if (ret != 0)
+        __wt_verbose_debug1(session, WT_VERB_LIVE_RESTORE,
+          "Live restore worker: Open cursor to %s ret %d", work_item->uri, ret);
     if (ret == ENOENT) {
         /* Free the work item. */
         __live_restore_free_work_item(session, &work_item);
         return (0);
+    } else if (ret == EBUSY) {
+        /*
+         * Someone else has exclusive access to the table. Add it back to the work queue and try
+         * again later.
+         */
+        __wt_spin_lock(session, &server->queue_lock);
+        TAILQ_INSERT_TAIL(&server->work_queue, work_item, q);
+        __wt_spin_unlock(session, &server->queue_lock);
+        return (0);
```

Comment:
If the last file in the queue is busy we'll have one final background thread burning CPU as it continually pops and reinserts the item.

```suggestion
        __wt_spin_unlock(session, &server->queue_lock);
        __wt_sleep(0, 100 * WT_THOUSAND); /* 0.1 seconds */
        return (0);
```
should fix it.

Reply:
I would only add this sleep if work_items remaining is 1 or something.

Reply:
Fixed. I did the "or something" where I sleep if work_items_remaining is less than the number of worker threads left. That way if more than one file is busy, many threads don't spin.